core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -22,7 +22,10 @@ namespace kagome::scale {
   struct TypeEncoder<std::array<T, sz>> {
     outcome::result<void> encode(const std::array<T, sz> &array,
                                  common::Buffer &out) {
-      return collection::encodeCollection(gsl::make_span(array), out);
+      for (int i = 0; i < sz; i++) {
```

Comment:
++i.

Reply:
Come on, why?

Reply:
theoretically, it helps produce faster code, well when optimization is off.

Reply:
I don't think that even if compiler doesn't optimize it one instruction less will save more time ever than we already spent discussing it.